src/mono/mono/eglib/CMakeLists.txt

```diff
@@ -67,3 +62,4 @@ target_include_directories(eglib_objects PRIVATE
   ${PROJECT_BINARY_DIR}/../..
   ${PROJECT_SOURCE_DIR})
 target_link_libraries (eglib_objects PUBLIC eglib_api)
+target_link_libraries(eglib_objects PUBLIC minipal)
```

Comment:
eglib doesn't need to expose minipal symbols through it. eglib_api exposes stuff for mono public API (src/native/public) for embedded, so it's different.

Reply:
This dependency is public as eglib_objects does depend on the minipal, so other places in Mono that use eglib_objects needs the minipal. That's why it's `PUBLIC`.

Reply:
> other places in Mono that use eglib_objects needs the minipal

Those other places being:

```
src/mono/mono/component/CMakeLists.txt
src/mono/mono/mini/CMakeLists.txt
src/mono/mono/profiler/CMakeLists.txt
```

can we not link minipal there as we have done in multiple coreclr projects?

Reply:
If we add it there, that's the exact same as adding it here, just more verbose and further from usage. We don't get any improvements by using `PRIVATE` in 4 places instead of `PUBLIC` in one.

I'm trying to move to using `PUBLIC` in more places where it's correct to reduce duplication and move our CMake to be more target-oriented instead of basically just throwing code at the wall and seeing what works. Mono's still quite a mess with the `TARGET_OBJECTS` usage, but at some point I'll clean that up.

Reply:
Overall, this PR is size improvement over main branch on NativeAOT side. Which means whatever was happening before with duplicate C sources in multiple projects, the linker is doing better job deduplicating the static libs in final binaries. Exposing those symbols via some other object would not impact the size. Mono is generally heavy on static libraries.

Reply:
`PUBLIC` doesn't expose symbols via different objects, it just specifies how dependencies are added to CMake targets. It's not a size concern in the final product, but a concern in the cleanliness and understandability of the infrastructure.

Reply:
If changing PUBLIC to PRIVATE and adding minipal to the three cmakelists files does not fix the issue, I understand that this is not the solution.

But if it does (due to the linkage details), what is the problem? I think it is rather 'better' this way because:
1. this is exactly what's been done in coreclr
2. is it an anti-pattern of sorts what we have in coreclr? I think this is debatable since we were thinking about cleaning up eglib during the utf8 work and calling minipal_utf8.. APIs directly where needed (mono/mini etc.). From that topology, it's currently a local concern of eglib that it make use of some minipal APIs. It is not exactly a 'complete PAL' for mono akin coreclr-pal.

Reply:
What I'm saying is I don't think changing `PUBLIC` to `PRIVATE` here would actually solve the problem.

The problem is that the `minipal_objects` library is linked into both the Mono runtime (which needs it to be in the static lib for when it statically links everything together for a native app) and into System.Native (which needs it as it depends on minipal as well). I don't want to try to thread the minipal static lib everywhere it needs to ship (see the brotli PR of mine that I still haven't finished due to this exact issue). In general, I'm trying to defer to linking the minipal objects into the runtime static libs and having the other static libs (that don't directly depend on the runtime) use the symbols from there.

This problem is unrelated to the `minipal` target mentioned here. The one mentioned here ensures that any Mono shared libs that depend on eglib_objects have the `minipal` static lib also linked into them as that's a transitive dependency.